c/driver/postgresql/statement.cc

```diff
@@ -255,6 +252,45 @@ struct BindStream {
 
   AdbcStatusCode Prepare(PGconn* conn, const std::string& query,
                          struct AdbcError* error) {
+    // tz-aware timestamps require special handling to set the timezone to UTC
+    // prior to sending over the binary protocol; must be reset after execute
+    for (int64_t col = 0; col < bind_schema->n_children; col++) {
+      if ((bind_schema_fields[col].type == ArrowType::NANOARROW_TYPE_TIMESTAMP) &&
+          (strcmp("", bind_schema_fields[col].timezone))) {
+        has_tz_field = true;
+
+        PGresult* begin_result = PQexec(conn, "BEGIN");
```

Comment:
Don't we need to hoist this out of the loop? Or else, only do this if `has_tz_field` was previously false. Otherwise, it looks like we'll try to begin the transaction twice.

(Also: I wonder if we shouldn't always begin/end a transaction when dealing with multiple bind parameters...)

Reply:
+how does this interact with auto-commit off?

Reply:
As far as autocommit goes I do think you are right about that being murky. I'm not sure how to best resolve that yet either - beginning / ending the transaction here within the `BindStream` class is probably at odds with most of the autocommit stuff being managed via the connection.

Are there tests already for autocommit in the ADBC suite? I was looking for a reference on how those are expected to behave. Wasn't sure if I was overlooking those of if it is something that hasn't been heavily implemented yet

Reply:
ah, I missed that little break at the end.

There are some simple tests that test the options, but not really the semantics.

At least here, I think you 'just' need to check if autocommit is off, if so, there's no need to BEGIN (you're already in a transaction), and you can still commit at the end (~though maybe semantically we shouldn't? This isn't a feature that's normally available in other APIs so I'm not sure what people expect)

Reply:
Any ideas on how to best forward the autocommit setting to the `BindStream` class? Right now I only see that as a private member of the connection, so it is pretty far removed from this

Reply:
A statement already holds a connection, so we can just add a getter to the PostgresConnection class